Wrappers/Python/test/test_algorithms.py

```diff
@@ -1161,55 +1160,39 @@ def test_SPDHG_vs_PDHG_implicit(self):
                     max_iteration=1000,
                     update_objective_interval=500)
-        pdhg.run(1000,verbose=0)
-
-        subsets = 10
-        size_of_subsets = int(len(angles)/subsets)
-        # take angles and create uniform subsets in uniform+sequential setting
-        list_angles = [angles[i:i+size_of_subsets]
-                       for i in range(0, len(angles), size_of_subsets)]
-        # create acquisitioin geometries for each the interval of splitting angles
-        list_geoms = [AcquisitionGeometry.create_Parallel2D().set_angles(list_angles[i], angle_unit='radian').set_panel(detectors, 0.1)
-                      for i in range(len(list_angles))]
-        # create with operators as many as the subsets
-        A = BlockOperator(*[ProjectionOperator(ig, list_geoms[i], dev)
-                          for i in range(subsets)])
-        # number of subsets
-        # (sub2ind, ind2sub) = divide_1Darray_equally(range(len(A)), subsets)
-        #
-        # acquisisiton data
-        AD_list = []
-        for sub_num in range(subsets):
-            for i in range(0, len(angles), size_of_subsets):
-                arr = noisy_data.as_array()[i:i+size_of_subsets, :]
-                AD_list.append(AcquisitionData(
-                    arr, geometry=list_geoms[sub_num]))
-
-        g = BlockDataContainer(*AD_list)
+        pdhg.run(100)
+        # %% 'explicit' SPDHG, scalar step-sizes
+        subsets = num_subsets
+        #partition the data
+        data_sub = noisy_data.partition(subsets, mode='staggered')
+        A = ProjectionOperator(ig, data_sub.geometry, dev)
 
         # block function
-        F = BlockFunction(*[KullbackLeibler(b=g[i]) for i in range(subsets)])
-        G = alpha * TotalVariation(50, 1e-4, lower=0, warm_start=True)
+        F = BlockFunction(*[KullbackLeibler(b=data_sub[i]) for i in range(subsets)])
+        G = alpha * TotalVariation(2, 1e-4, lower=0, warm_start=True)
 
         prob = [1/len(A)]*len(A)
         spdhg = SPDHG(f=F, g=G, operator=A,
-                      max_iteration=1000,
-                      update_objective_interval=200, prob=prob)
-        spdhg.run(1000, verbose=0)
+                        update_objective_interval=200, prob=prob)
+        spdhg.run(20 * subsets)
+
         qm = (mae(spdhg.get_output(), pdhg.get_output()),
               mse(spdhg.get_output(), pdhg.get_output()),
               psnr(spdhg.get_output(), pdhg.get_output())
               )
         log.info("Quality measures %r", qm)
 
         np.testing.assert_almost_equal(mae(spdhg.get_output(), pdhg.get_output()),
-                                       0.000335, decimal=3)
+                                       0.00189242, decimal=3)
         np.testing.assert_almost_equal(mse(spdhg.get_output(), pdhg.get_output()),
-                                       5.51141e-06, decimal=3)
+                                       3.9063532e-05, decimal=3)
```

Comment:
Where do these numbers come from? Do we want the error to go to zero?

Reply:
these numbers come from those measures that I printed out. It is an awful system, but at least this time I visually inspected the 2 solutions!